### PR TITLE
[Keyboard] Add ReverseStudio-DecadePad

### PR DESCRIPTION
**Pull request by huajijam**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/ReverseStudio/DecadePad/DecadePad.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include "quantum.h"
+#define LAYOUT( \
```
Comment (drashna) [inline keyboards/ReverseStudio/DecadePad/DecadePad.h]: This is actually a supported layout for the community layouts. 

```suggestion

#define LAYOUT_numpad_6x4( \
```

File: keyboards/ReverseStudio/DecadePad/keymap.c [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+[0] = LAYOUT(
```
Comment (drashna) [inline keyboards/ReverseStudio/DecadePad/keymap.c]:
```suggestion
[0] = LAYOUT_numpad_6x4(

```

File: keyboards/ReverseStudio/DecadePad/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+[0] = LAYOUT(
```
Comment (drashna) [inline keyboards/ReverseStudio/DecadePad/keymaps/default/keymap.c]:
```suggestion
[0] = LAYOUT_numpad_6x4(

```

File: keyboards/ReverseStudio/DecadePad/rules.mk [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+AUDIO_ENABLE = no           # Audio output on port C6
+UNICODE_ENABLE = yes         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+RGBLIGHT_ENABLE = yes       # Enable WS2812 RGB underlight.
+API_SYSEX_ENABLE = no
+
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+
+LAYOUTS = default
```
Comment (drashna) [inline keyboards/ReverseStudio/DecadePad/rules.mk]:
```suggestion
LAYOUTS = numpad_6x4
```

File: keyboards/ReverseStudio/DecadePad/keymap.c [hunk 5]
```diff
@@ -0,0 +1,12 @@
+#include QMK_KEYBOARD_H
```
Comment (zvecr) [inline keyboards/ReverseStudio/DecadePad/keymap.c]: This file should be deleted

File: keyboards/ReverseStudio/DecadePad/layout.json [hunk 6]
```diff
@@ -0,0 +1,7 @@
+[{a:7},"","","",""],
```
Comment (zvecr) [inline keyboards/ReverseStudio/DecadePad/layout.json]: This file should be deleted

**Comment by huajijam** [comment]
thx guys, it's done now

**Comment by zvecr** [comment]
Community layouts would be supported by this, not sure what you had as the commits have now gone.

**Comment by huajijam** [comment]
Now the firmware have bugs
I don't know why Col2Row5 at board will be Col1Row5,and Col3Row5 will be Col2Row5
I am trying to revert the version back,but it doesn't succes

**Comment by huajijam** [comment]
Community layouts is worked, but the col problem is still, it just happen between row5col2 and col3,looks like my firmware matrix is wrong, i try to build a firmware at https://kbfirmware.com/ and it works,
image

**Comment by huajijam** [comment]
is there any wrong with my firmware matrix?

**Comment by huajijam** [comment]
All bugs have been fix,thx guys

File: keyboards/reversestudio/decadepad/info.json [hunk 7]
```diff
@@ -0,0 +1,38 @@
+{
+    "keyboard_name": "DecadePad",
+    "url": "",
+    "maintainer": "huajijam",
+    "width": 6,
+    "height": 4,
+    "layouts": {
+        "LAYOUT": {
```
Comment (zvecr) [inline keyboards/reversestudio/decadepad/info.json]:
```suggestion
        "LAYOUT_numpad_6x4": {
```

File: keyboards/reversestudio/decadepad/rules.mk [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+UNICODE_ENABLE = yes         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+RGBLIGHT_ENABLE = yes       # Enable WS2812 RGB underlight.
+API_SYSEX_ENABLE = no
+
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+
+LAYOUTS = numpad_6x4
+LAYOUTS_HAS_RGB = no
```
Comment (zvecr) [inline keyboards/reversestudio/decadepad/rules.mk]:
```suggestion
LAYOUTS = numpad_6x4
```

File: keyboards/reversestudio/decadepad/decadepad.c [hunk 9]
```diff
@@ -0,0 +1,12 @@
+#include "decadepad.h"
+bool led_update_kb(led_t led_state) {
+    if (led_update_user(led_state)){
+        // writePin sets the pin high for 1 and low for 0.
+        // In this example the pins are inverted, setting
+        // it low/0 turns it on, and high/1 turns the LED off.
+        // This behavior depends on whether the LED is between the pin
+        // and VCC or the pin and GND.
```
Comment (fauxpark) [inline keyboards/reversestudio/decadepad/decadepad.c]:
```suggestion
```

File: keyboards/reversestudio/decadepad/keymap.c [hunk 10]
```diff
@@ -0,0 +1,12 @@
+#include QMK_KEYBOARD_H
```
Comment (fauxpark) [inline keyboards/reversestudio/decadepad/keymap.c]: Please delete this file.

File: keyboards/reversestudio/decadepad/keymaps/default/keymap.c [hunk 11]
```diff
@@ -0,0 +1,12 @@
...
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [0] = LAYOUT_numpad_6x4(
+    RGB_TOG,  RGB_MOD,  RGB_VAI,  RGB_M_R,
+    KC_NLCK,  KC_PSLS,  KC_PAST,  KC_PMNS,
+    KC_P7,    KC_P8,    KC_P9,        
+    KC_P4,    KC_P5,    KC_P6,    KC_PPLS,
+    KC_P1,    KC_P2,    KC_P3,    
+    KC_P0,              KC_PDOT,  KC_PENT        
+  )
```
Comment (fauxpark) [inline keyboards/reversestudio/decadepad/keymaps/default/keymap.c]:
```suggestion
    [0] = LAYOUT_numpad_6x4(
        RGB_TOG,  RGB_MOD,  RGB_VAI,  RGB_M_R,
        KC_NLCK,  KC_PSLS,  KC_PAST,  KC_PMNS,
        KC_P7,    KC_P8,    KC_P9,
        KC_P4,    KC_P5,    KC_P6,    KC_PPLS,
        KC_P1,    KC_P2,    KC_P3,
        KC_P0,              KC_PDOT,  KC_PENT
    )
```

File: keyboards/reversestudio/readme.md [hunk 12]
```diff
@@ -0,0 +1 @@
+Made By huajijam
```
Comment (fauxpark) [inline keyboards/reversestudio/readme.md]: Could you please replace this with the readme from the template?

File: keyboards/reversestudio/decadepad/rules.mk [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   change to "no" to disable the options, or define them in the Makefile in
+#   the appropriate keymap folder that will get included automatically
```
Comment (fauxpark) [inline keyboards/reversestudio/decadepad/rules.mk]:
```suggestion
#   change yes to no to disable
```

File: keyboards/reversestudio/decadepad/readme.md [hunk 14]
```diff
@@ -0,0 +1,10 @@
+# DecadePad
+A 6x4 Standard Numpad
+
+* Keyboard Maintainer: [huajijam](https://github.com/huajijam)
+* Hardware Supported: DecadePad
+Make example for this keyboard (after setting up your build environment):
```
Comment (fauxpark) [inline keyboards/reversestudio/decadepad/readme.md]:
```suggestion

Make example for this keyboard (after setting up your build environment):
```

File: keyboards/reversestudio/decadepad/readme.md [hunk 15]
```diff
@@ -0,0 +1,10 @@
+# DecadePad
+A 6x4 Standard Numpad
```
Comment (fauxpark) [inline keyboards/reversestudio/decadepad/readme.md]:
```suggestion

A 6x4 Standard Numpad
```

File: keyboards/reversestudio/decadepad/info.json [hunk 16]
```diff
@@ -0,0 +1,38 @@
...
+                {"x":2, "y":1},
+                {"x":2, "y":2},
+                {"x":2, "y":3},
+                {"x":2, "y":4},
+                {"x":2, "y":5},
+
+                {"x":3, "y":0},
+                {"x":3, "y":1},
+                {"x":3, "y":2, "h":2},
+                {"x":3, "y":4, "h":2},
```
Comment (noroadsleft) [inline keyboards/reversestudio/decadepad/info.json:34]: These objects need to be in the same order that `keymap.c` and `decadepad.h` use.

```suggestion
                {"x":0, "y":0},
                {"x":1, "y":0},
                {"x":2, "y":0},
                {"x":3, "y":0},

                {"x":0, "y":1},
                {"x":1, "y":1},
                {"x":2, "y":1},
                {"x":3, "y":1},

                {"x":0, "y":2},
                {"x":1, "y":2},
                {"x":2, "y":2},

                {"x":0, "y":3},
                {"x":1, "y":3},
                {"x":2, "y":3},
                {"x":3, "y":2, "h":2},

                {"x":0, "y":4},
                {"x":1, "y":4},
                {"x":2, "y":4},

                {"x":0, "y":5, "w":2},
                {"x":2, "y":5},
                {"x":3, "y":4, "h":2}
```

File: keyboards/reversestudio/decadepad/config.h [hunk 17]
```diff
@@ -0,0 +1,92 @@
...
+
+
+//via:
+
+// Dynamic keymap starts after EEPROM version
+#define DYNAMIC_KEYMAP_EEPROM_ADDR 35
+// Dynamic macro starts after dynamic keymaps (35+(4*10*6*2)) = (35+480)
+#define DYNAMIC_KEYMAP_MACRO_EEPROM_ADDR 227     // **** CHANGE THIS BASED ON MATRIX_ROWS & MATRIX_COLS ****
+#define DYNAMIC_KEYMAP_MACRO_EEPROM_SIZE 797     // **** CHANGE THIS BASED ON 1024-DYNAMIC_KEYMAP_MACRO_EEPRO   _ADDR ****
+#define DYNAMIC_KEYMAP_MACRO_COUNT 16
```
Comment (noroadsleft) [inline keyboards/reversestudio/decadepad/config.h:92]: You shouldn't need any of this to run VIA.

```suggestion
```

File: keyboards/reversestudio/decadepad/keymaps/via/keymap.c [hunk 18]
```diff
@@ -0,0 +1,12 @@
...
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [0] = LAYOUT_numpad_6x4(
+      RGB_TOG,  RGB_MOD,  RGB_VAI,  RGB_M_R,
+      KC_NLCK,  KC_PSLS,  KC_PAST,  KC_PMNS,
+      KC_P7,    KC_P8,    KC_P9,
+      KC_P4,    KC_P5,    KC_P6,    KC_PPLS,
+      KC_P1,    KC_P2,    KC_P3,    
+      KC_P0,              KC_PDOT,  KC_PENT
+  )
```
Comment (noroadsleft) [inline keyboards/reversestudio/decadepad/keymaps/via/keymap.c:11]: VIA by default expects a minimum of four layers; not including four can lead to odd behaviour when the use goes to edit their keymap in VIA Configurator.

```suggestion
  ),
  [1] = LAYOUT_numpad_6x4(
      KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,
      KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,
      KC_TRNS,  KC_TRNS,  KC_TRNS,
      KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,
      KC_TRNS,  KC_TRNS,  KC_TRNS,    
      KC_TRNS,            KC_TRNS,  KC_TRNS
  ),
  [2] = LAYOUT_numpad_6x4(
      KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,
      KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,
      KC_TRNS,  KC_TRNS,  KC_TRNS,
      KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,
      KC_TRNS,  KC_TRNS,  KC_TRNS,    
      KC_TRNS,            KC_TRNS,  KC_TRNS
  ),
  [3] = LAYOUT_numpad_6x4(
      KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,
      KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,
      KC_TRNS,  KC_TRNS,  KC_TRNS,
      KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,
      KC_TRNS,  KC_TRNS,  KC_TRNS,    
      KC_TRNS,            KC_TRNS,  KC_TRNS
  ) 
```

File: keyboards/reversestudio/decadepad/keymaps/via/rule.mk [hunk 19]
```diff
@@ -0,0 +1,2 @@
+VIA_ENABLE = yes
```
Comment (noroadsleft) [inline keyboards/reversestudio/decadepad/keymaps/via/rule.mk:1]: This file needs to be named `rules.mk` in order to work.

File: keyboards/reversestudio/decadepad/rules.mk [hunk 20]
```diff
@@ -0,0 +1,33 @@
...
+BACKLIGHT_ENABLE = no      # Enable keyboard backlight functionality
+MIDI_ENABLE = no            # MIDI controls
+AUDIO_ENABLE = no           # Audio output on port C6
+UNICODE_ENABLE = yes         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+RGBLIGHT_ENABLE = yes       # Enable WS2812 RGB underlight.
+API_SYSEX_ENABLE = no
+
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
```
Comment (noroadsleft) [inline keyboards/reversestudio/decadepad/rules.mk:31]: Edits this section so it aligns to QMK's template; no logic or settings changes here.

```suggestion
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = yes      # Virtual DIP switch configuration
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = yes        # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover 
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
MIDI_ENABLE = no            # MIDI support
BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
AUDIO_ENABLE = no           # Audio output on port C6
FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
UNICODE_ENABLE = yes        # Unicode
```

**Comment by huajijam** [comment]
thx guys, via is finally enable now

**Pull request merged**